core/src/docs/rfcs/5485_options_for_reader_with.md

```diff
@@ -0,0 +1,89 @@
+- Proposal Name: `options_for_reader_with`
+- Start Date: 2024-12-31
+- RFC PR: [apache/opendal#5485](https://github.com/apache/opendal/pull/5485)
+- Tracking Issue: [apache/opendal#5486](https://github.com/apache/opendal/issues/5486)
+
+# Summary
+
+Add `if_modified_since` and `if_unmodified_since` options to OpenDAL's `reader_with` API.
+
+# Motivation
+
+OpenDAL currently supports conditional `reader_with` operations based only on `version`. However, many storage services also 
+support conditional operations based on modification time.
+
+Adding these options will:
+
+- Provide more granular control over read operations.
+- Align OpenDAL with features provided by modern storage services, meeting broader use cases.
+
+# Guide-level explanation
+
+Two new options will be added to the `reader_with` API:
```

Comment:
Let's complete the RFC directly by including `if-match` and `if-none-match` directly.